### PR TITLE
chore: v1.7 repo organization & template hardening

### PR DESCRIPTION
**Pull request by jml6m**
## What

v1.7 integration branch — repo organization, template hardening, dead-asset cleanup, and security gate improvements ahead of v2 template/rule lockdown.

No version bump. No public API changes. Topic PRs stack on `chore/v1.7-repo-org`.

## Why

v1.6 shipped languages theorem protocol + security gates. v1.7 is housekeeping: one source of truth for agent/contributor docs, retire Phase 2/3 prep cruft, align CI/templates with the current branch model (`master` + per-release integration branches), and harden the release path with explicit security gates.

### `.github/` — governance & CI
- `copilot-instructions.md` — synced from `AGENTS.md`; removed stale Browserify/Flow/nyc refs
- `pull_request_template.md` — updated; removed `main-v1-1-prep`, Flow, 100% coverage refs
- `dependabot.yml` — all intervals set to `monthly` (no interval less than monthly)
- `ci.yml` — updated push branch patterns; `security` job kept separate (npm audit is a distinct concern from test correctness)
- `ISSUE_TEMPLATE/epic.yml` — replaced hardcoded `v1.x`/`v2` phase options with version-agnostic categories (Governance/tooling/CI, Demo/UX/docs, Public API — FSA operations, Language algorithms — internal, Test infrastructure, Security/dependencies, Cross-cutting)
- `publish.yml` — added `check-package-scripts` + `check:security` gates before `npm publish`, matching the CI security model

### `docs/` — legacy prep docs removed
- Deleted `docs/v1.1-prep/README.md` and `docs/v1.1-prep/test-architecture.md` (superseded; live references are `AGENTS.md` and `CONTRIBUTING.md`)
- Deleted `docs/languages-testing.md` (test file is self-documenting)
- Deleted `docs/types-and-correctness.md` — content inlined directly into `AGENTS.md` and `copilot-instructions.md` under the "Testing" section
- Updated `docs/v1.1-prep/coverage-policy.md` to remove dead links to deleted files
- `AGENTS.md`, `CONTRIBUTING.md`, `copilot-instructions.md` — updated all references to deleted files; dependabot interval note updated to monthly

### `test/` — instanceOf coverage improved
- `test/utils.spec.js` — added positive happy-path tests for real DFA/NFA instances; simplified negative tests to meaningful cases (nullish/primitive, wrong class, unrelated type, hostile proxy); removed 3 overly specific proxy-introspection tests; corrected assertion direction (`NFA extends DFA` so `instanceOf(DFA, nfa)` correctly returns `true`)

### `codecov.yml` — audited
- Removed no-op `patch` section (`target: auto, threshold: 0%, informational: true` enforced nothing)
- `ignore` and `component_management` sections confirmed correct and unchanged

## How tested

- [x] `npm ci`
- [x] `npm test` — 210 passing, coverage ≥ 90% on all metrics (statements, branches, functions, lines)

## Checklist

- [x] PR targets `master` (integration → stable release PR)
- [x] No version bump in `package.json`
- [x] Tests added or updated for new/changed behavior (`test/utils.spec.js` instanceOf improvements)
- [x] Test Coverage maintained
- [ ] Child topic PRs target `chore/v1.7-repo-org`

### CODE REVIEW

**Comment by jml6m** [comment]
Update for wrap-up: PR #259 has been closed (superseded). The key fixes from it (base-SHA config loading for bootstrapping the new lock gate, additional test coverage, hardening) have been condensed into this branch (see recent local commit on chore/v1.7-repo-org). 

Full verification: typecheck/lint/build ✅, npm test (196 passing, coverage >=90% floor) ✅, check:security ✅. No stale theorem refs remain in publish or key paths. ci.yml/publish.yml aligned (triggers + npm upgrade in CI, publish is now bare-min calling npm ci).

Local state prepped and ready (ahead of origin by condense commit). Ready for final review/merge to master when appropriate. (Changes respect v1.7 branch workflow and prior cleanups.)

**Comment by jml6m** [comment]
259 closed as superseded (see prior comment). Local chore/v1.7-repo-org is now at the condensed state with all lock/259 fixes + ci/publish alignment. Full 
pm test green (196 pass, coverage floor met). Branch ready for review. (Note: local is 1 ahead of origin due to condense commits; no push from agent side.)

File: scripts/check-protected-files.mjs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+export function loadPatterns() {
+  const configPath = resolve(root, ".github/PROTECTED_FILES.json");
+  const config = JSON.parse(readFileSync(configPath, "utf8"));
+  if (!Array.isArray(config.patterns) || config.patterns.length === 0) {
+    throw new Error(".github/PROTECTED_FILES.json must define a non-empty patterns array");
+  }
+  return config.patterns.map(globToRegExp);
+}
```
[the remark this answers is not in the document]
Comment (jml6m) [reply scripts/check-protected-files.mjs]: @copilot agreed, proceed with fix

File: package.json [hunk 2]
```diff
@@ -25,18 +25,18 @@
...
+
     "free-port": "node scripts/free-port.mjs 3200",
     "preserve:demo": "node scripts/free-port.mjs 3200",
     "serve:demo": "node scripts/serve-demo.mjs",
     "test:demo": "npm run build && cross-env NODE_OPTIONS=--import=tsx mocha \"test/demo.spec.js\"",
     "test:presets": "cross-env NODE_OPTIONS=--import=tsx mocha \"test/demo-presets-golden.spec.js\"",
     "test:languages": "cross-env NODE_OPTIONS=--import=tsx mocha \"test/languages.spec.js\"",
-    "test": "npm run guard:theorem && npm run typecheck && npm run lint && npm run build && npm run check:security && cross-env NODE_OPTIONS=--import=tsx c8 mocha \"test/**/*.spec.js\"",
+    "test": "npm run typecheck && npm run lint && npm run build && npm run check:security && cross-env NODE_OPTIONS=--import=tsx c8 mocha \"test/**/*.spec.js\"",
     "audit:ci": "npm audit --audit-level=high",
```
[the remark this answers is not in the document]
Comment (jml6m) [reply package.json:39]: @copilot we don't need that script or guard, make necessary adjustments on this PR based on that feedback.

File: src/languages/NFAtoDFA.ts [hunk 3]
```diff
@@ -1,7 +1,6 @@
 import { type State } from "../components/State";
-import { type FSA } from "../interfaces/FSA";
+import type { FSA } from "../interfaces/FSA";
 import { NFA } from "../automata";
-import { instanceOf } from "../globals/globals";
 import type { TransitionInput } from "../utils/DFAUtils";
 import { languageAlphabetSymbols } from "./fsaHelpers";
```
[the remark this answers is not in the document]
Comment (jml6m) [reply src/languages/NFAtoDFA.ts:5]: @copilot What is you're recommended fix?  include some code snippets (pseudocode or small test samples) that would help me understand.

File: src/languages/NFAtoDFA.ts [hunk 4]
```diff
@@ -67,13 +66,9 @@ function subsetWitness(members: State[]): readonly string[] {
...
- * @fas-spec Powerset (subset) construction with ε-closure (Sipser Thm. 1.19).
- */
-/* @coverage-caveat: c8 line hits here mean fixture NFA instances were exercised — not Σ* verification */
-export function subsetConstruction(nfa: FSA): SubsetConstructionResult {
-  if (!instanceOf(NFA, nfa)) {
+/** Powerset (subset) construction with ε-closure. Requires an NFA (enforced by type and runtime class check). */
+export function subsetConstruction(nfa: NFA): SubsetConstructionResult {
+  if (!(nfa instanceof NFA)) {
     throw new TypeError("subsetConstruction requires an NFA");
   }
```
[the remark this answers is not in the document]
Comment (jml6m) [reply src/languages/NFAtoDFA.ts:78]: @copilot Do you agree with this point? I'm having trouble committing the changes, maybe it's a locked file?

**Review by jml6m — COMMENTED** [review]
@copilot see discussion comments

**Comment by jml6m** [comment]
> @jml6m I'm unable to start working on this because of repository rules that prevent me from pushing to the branch:
> 
> * Changes must be made through a pull request due to repository rules
> 
> See [the documentation](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/troubleshooting-rules) for more details.

@copilot in that case open up a PR on top of this branch for the purposes of code review fixes

File: .github/ISSUE_TEMPLATE/epic.yml [hunk 5]
```diff
@@ -38,10 +38,8 @@ body:
       label: Phase
       description: Which project phase does this epic belong to?
       options:
```
Comment (jml6m) [inline .github/ISSUE_TEMPLATE/epic.yml:40]: @copilot let's rethink these categories...first off, no hardcoded numbers in the v*.* patterns (options should always be agnostic of actual numbers because the enhancements will go on beyond 1 and 2)

Also the categories should more closely align with how the repo is structured, and how the app is consumed (via public apis, and your knowledge of the academic area)

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -32,13 +37,21 @@ jobs:
...
+        run: npm install -g npm@^11.5.1
       - run: npm ci
+      # Run directly (not via npm scripts) so package.json script tampering
+      # cannot skip the integrity gate before npm test executes.
+      - run: node scripts/check-package-scripts.mjs
       - run: npm test
       - name: Upload coverage to Codecov
         if: matrix.node == 20
-        uses: codecov/codecov-action@e79a6962e0d4c0c17b229090214935d2e33f8354 # v6.0.1
+        uses: codecov/codecov-action@fb8b3582c8e4def4969c97caa2f19720cb33a72f # v7.0.0
```
Comment (jml6m) [inline .github/workflows/ci.yml:52]: @copilot why is there a long hash here on this line? Is it something that will constantly need changing? That doesn't seem ideal

File: .github/workflows/ci.yml
Comment (jml6m) [inline .github/workflows/ci.yml:58]: @copilot Thinking about it now, is there a reason this is a separate job? If you look at all the use cases, does the structure of this file hold up?

File: .github/workflows/publish.yml [hunk 7]
```diff
@@ -22,13 +22,7 @@ jobs:
...
-      # is violated. Runs before install so a bad tag fails fast.
-      - run: node scripts/check-theorem-annotations.mjs
-      # npm CLI 10 (shipped with Node 20) does not fully support npm Trusted
-      # Publishing OIDC; the registry PUT falls through unauthenticated and
-      # returns a misleading 404. Upgrade to npm 11.5.1+ before publishing.
+      # Use recent npm for OIDC provenance support (see ci.yml for rationale).
       - run: npm install -g npm@^11.5.1
       - run: npm ci
-      - run: npm test
       - run: npm publish --provenance --access public
```
[the remark this answers is not in the document]
Comment (jml6m) [reply .github/workflows/publish.yml:30]: @copilot yes update this file based on our slew of improvements to the security of this repo and other workflows/files

File: .github/dependabot.yml
Comment (jml6m) [inline .github/dependabot.yml]: @copilot ideally we don't have anything less than monthly on any interval in dependabot.yml if you want to audit that, make changes if possible, and update relevant .md instructions for agents

File: demo/README.md
Comment (jml6m) [inline demo/README.md:1]: @copilot further check this file by: removing hardcoded values wherever possible that shouldn't really be there, make it applicable to all versions up to v1.7 (include only important information), remove information that's no longer relevant, do another audit, etc.

File: docs/v1.1-prep/README.md
Comment (jml6m) [inline docs/v1.1-prep/README.md:1]: @copilot ideally this file is deleted. If some of the information here is still needed, let's discuss.

File: docs/v1.1-prep/test-architecture.md
Comment (jml6m) [inline docs/v1.1-prep/test-architecture.md:1]: @copilot  ideally this file is deleted. If some of the information here is still needed, let's discuss.

File: docs/languages-testing.md
Comment (jml6m) [inline docs/languages-testing.md:1]: @copilot This file doesn't seem needed, some of this is better put in code comments, if necessary. Otherwise doesn't seem that needed after all our refactors/optimizations

File: docs/types-and-correctness.md
Comment (jml6m) [inline docs/types-and-correctness.md:1]: @copilot why is this its own file and not part of another file?

File: src/languages/NFAtoDFA.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (jml6m) [reply src/languages/NFAtoDFA.ts:78]: @copilot make relevant updates to this function (and this file) based on our new updates to `instanceOf` and other global/helper functions

File: test/utils.spec.js
Comment (jml6m) [inline test/utils.spec.js:102]: @copilot for the `instanceOf` tests, particularly the ones that return false, ideally we have something more generalized for that. I forget exactly what triggered adding these types of very granular object prototype-esque tests, i think there is some discussiong in #260 or #261 if you look over all the discussion and code review comments.

I was probably concerned about something changing in the lower level code, some scenarios that would be need to be tested. Look through the discussion history to see if we can identify what really needs to be tested here, both positive (I don't see many, or any, happy path positive tests for `instanceOf`) and negative (ideally there's probably some simpler tests than these really specific ones)

File: codecov.yml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+coverage:
+  status:
+    project:
+      default:
+        target: 90%
+        threshold: 1%
+        informational: true
+        if_ci_failed: error
+    patch:
```
Comment (jml6m) [inline codecov.yml]: @copilot why is `patch` used here?

File: codecov.yml [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+      default:
+        target: auto
+        threshold: 0%
+        informational: true
+
+comment: false
+
+ignore:
```
Comment (jml6m) [inline codecov.yml:15]: @copilot do another audit for this section, make sure it's exactly what we need.

File: codecov.yml [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  - "src/interfaces/**"
+  - "src/**/index.ts"
+  - "src/demo-bundle.ts"
+
+component_management:
```
Comment (jml6m) [inline codecov.yml:20]: @copilot do another audit of this section as well (and explain your reasoning)

File: package.json [hunk 11]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "fas-js",
   "description": "Finate State Automata JS Solutions",
-  "version": "1.6.0",
+  "version": "1.7.0",
   "type": "module",
```
[the remark this answers is not in the document]
Comment (jml6m) [reply package.json:5]: @copilot correct, version bumps happen after a qualified branch is merged in (I think)

File: package.json [hunk 12]
```diff
@@ -25,19 +25,19 @@
...
     "preserve:demo": "node scripts/free-port.mjs 3200",
     "serve:demo": "node scripts/serve-demo.mjs",
     "test:demo": "npm run build && cross-env NODE_OPTIONS=--import=tsx mocha \"test/demo.spec.js\"",
     "test:presets": "cross-env NODE_OPTIONS=--import=tsx mocha \"test/demo-presets-golden.spec.js\"",
     "test:languages": "cross-env NODE_OPTIONS=--import=tsx mocha \"test/languages.spec.js\"",
-    "test": "npm run guard:theorem && npm run typecheck && npm run lint && npm run build && npm run check:security && cross-env NODE_OPTIONS=--import=tsx c8 mocha \"test/**/*.spec.js\"",
+    "test": "npm run typecheck && npm run lint && npm run build && npm run check:security && cross-env NODE_OPTIONS=--import=tsx c8 mocha \"test/**/*.spec.js\"",
     "audit:ci": "npm audit --audit-level=high",
+    "check:package-scripts": "node scripts/check-package-scripts.mjs",
     "check:security": "node scripts/check-public-api.mjs && node scripts/check-npm-pack.mjs",
```
Comment (jml6m) [inline package.json:41]: @copilot I'm still wondering if this is correct. Should `node scripts/check-package-scripts.mjs` still be included in this command AND available in `check:package-scripts` for the one off situations where it's needed for workflows?

I just want to make sure we're closing all gaps.

**Review by jml6m — COMMENTED** [review]
@copilot lots of code review comments to address

**Comment by jml6m** [comment]
@copilot All 210 tests pass, coverage ≥ 90%. You can apply the commit by pulling from this local state or cherry-picking 56194d3.

can you give me the exact command i should via command line with project admin?

**Comment by jml6m** [comment]
@copilot  All 210 tests pass, coverage ≥ 90%. You can apply the commit by pulling from this local state or cherry-picking https://github.com/jml6m/fas-js/commit/56194d38e86b45e37b905070a453b04cc0e91dd2.

can you give me the exact command i should via command line with project admin?
